File: Seapal_Gruppe_09/de.htwg.seapal-master/de.htwg.seapal-master/Seapal_php/js/app/ajax/trackingpoint.js
$(function() {
	
	function loadEntry(id) { 
			        	
	    jQuery.get("app_tracking_point_load.php", {'trackpointnr': id}, function(data) {
	        $('#strength').val(data['windstaerke']);
	        $('#wind_direction').val(data['windrichtung']);
	        $('#airpressure').val(data['luftdruck']);
	        $('#temperature').val(data['temperatur']);
	        $('#clouds').val(data['wolken']);
	        $('#rain').val(data['regen']);
	        $('#waveheight').val(data['wellenhoehe']);
	        $('#wave_direction').val(data['wellenrichtung']);
	        $('#marker').val(data['marker']);
	        $('#btm').val(data['btm']);
	        $('#dtm').val(data['dtm']);
	        $('#sog').val(data['sog']);
	        $('#cog').val(data['cog']);
	        $('#manoever').val(data['manoever']);
	        $('#vorsegel').val(data['vorsegel']);
	        $('#weather_date').val(data['wdate']);
	        $('#weather_time').val(data['wtime']);
	        $('#motor').val(data['motor']);
	        $('#tank').val(data['tank']);
	    } , "json");
	}
	
	function addEntry(trackpointnr, json) {
		
		var entry = ""; 
			
		entry += "<tr class='selectable' id='" + trackpointnr + "'>";
		entry += "<td>" + json.marker + "</td>";
		entry += "<td>" + json.btm + "</td>";
		entry += "<td>" + json.dtm + "</td>";
		entry += "<td>" + json.sog + "</td>";
		entry += "<td>" + json.cog + "</td>";
		entry += "<td>" + json.manoever + "</td>";
		entry += "<td>" + json.vorsegel + "</td>";
		entry += "<td>" + json.wdate + "</td>";
		entry += "<td>" + json.wtime + "</td>";
		entry += "<td>" + json.motor + "</td>";
		entry += "<td>" + json.tank + "</td>";
        entry += "<td>" + json.strength + "</td>";
        entry += "<td>" + json.wind_direction + "</td>";
        entry += "<td>" + json.airpressure + "</td>";
        entry += "<td>" + json.temperature + "</td>";
        entry += "<td>" + json.clouds + "</td>";
        entry += "<td>" + json.rain + "</td>";
        entry += "<td>" + json.wave_direction + "</td>";
        entry += "<td>" + json.waveheight + "</td>";
        entry += "<td style='width:30px; text-align:left;'><div class='btn-group'>";
        entry += "<a class='btn btn-small view trackingpoint' id='" + trackpointnr + "'><span><i class='icon-eye-open'></i></span></a>";
        entry += "<a class='btn btn-small remove trackingpoint' id='" + trackpointnr + "'><span><i class='icon-remove'></i></span></a>";
        entry += "</div></td>";
        entry += "</tr>";

		$('#entries_trackingpoint').append(entry);
	}

	$('a.view.trackingpoint').live("click", function(event) {
		loadEntry($(this).attr('id'));
	});

	$('a.remove.trackingpoint').live("click", function(event) {
		var buttonID = this;
	 	var id = $(this).attr('id');
		jQuery.post("app_tracking_point_delete.php", { "trackpointnr": id }, function(data) {
		 
		 	if (data['trackpointnr'].match(/Error/)) {
		    	
		    	$('#dialogTitle').text('Error');
		    	$('#dialogMessage').text(data['trackpointnr'].replace(/Error: /, ""));
		    	
	    	} else {
		    	
		    	$(buttonID).parents('tr').remove();  
	    
		    	$('#dialogTitle').text('Succes');
		    	$('#dialogMessage').text("Eintrag wurde erfolgreich gelöscht.");
	    	}
			
			$('#messageBox').modal('show');
		}, "json");
	});

	$('#save_trackingpoint').click(function(event) {	
		event.preventDefault();

		var query = window.location.search;
		
		var tracknrQuery = query.match(/tracknr=\d/);
		var tracknr = tracknrQuery[0].replace(/tracknr=/, "");
		
		var json = {
				"tracknr": tracknr,
	          	"strength": $('#strength').val(),
	          	"wind_direction": $('#wind_direction').val(),
	         	"airpressure": $('#airpressure').val(),
		        "temperature": $('#temperature').val(),
		        "clouds": $('#clouds').val(),
		        "rain": $('#rain').val(),
		        "waveheight": $('#waveheight').val(),
		        "wave_direction": $('#wave_direction').val(),
		        "marker": $('#marker').val(),
		        "btm": $('#btm').val(),
		        "dtm": $('#dtm').val(),
		        "sog": $('#sog').val(),
		        "cog": $('#cog').val(),
		        "manoever": $('#manoever').val(),
		        "vorsegel": $('#vorsegel').val(),
		        "wdate": $('#weather_date').val(),
		        "wtime": $('#weather_time').val(),
		        "motor": $('#motor').val(),
		        "tank": $('#tank').val(),
	    };

	    console.log(json);
		
		jQuery.post("app_tracking_point_insert.php", json, function(data) { 
			
			if (data['trackpointnr'].match(/Error/)) {
		    	
		    	$('#dialogTitle').text('Error');
		    	$('#dialogMessage').text(data['trackpointnr'].replace(/Error: /, ""));
		    	
	    	} else {
		    	
		    	addEntry( data['trackpointnr'], json ); 
	    
		    	$('#dialogTitle').text('Success');
		    	$('#dialogMessage').text("Eintrag wurde erfolgreich gespeichert.");
	    	}
			
			$('#messageBox').modal('show');
		
		}, "json");
	
	});

});

File: Seapal_Gruppe_09/de.htwg.seapal.detp-master/Seapal/app/controllers/Trackingpoint.java
package controllers;

import play.*;
import play.mvc.*;
import play.db.*;
import java.sql.*;
import javax.sql.*;
import play.libs.Json;
import play.data.DynamicForm;
import org.codehaus.jackson.node.ObjectNode; 
import views.html.*;
import views.html._include.*;

public class Trackingpoint extends Controller {
  
	public static Result index(int tracknr) {  
		Connection conn = DB.getConnection();
		String responseData = "";

		if(conn != null) {
			Statement query;
			ResultSet result;
			StringBuilder row;

			try {
				query = conn.createStatement();
				String sql = "SELECT * " + "FROM seapal.trackingpoint WHERE tracknr = " + tracknr;
				result = query.executeQuery(sql);

				while (result.next()) {
					row = new StringBuilder();
					
					row.append("<tr class='selectable' id='" + result.getString("trackpointnr") + "'>");
					row.append("<td>" + result.getString("marker") + "</td>");
					row.append("<td>" + result.getString("btm") + "</td>");
					row.append("<td>" + result.getString("dtm") + "</td>");
					row.append("<td>" + result.getString("sog") + "</td>");
					row.append("<td>" + result.getString("cog") + "</td>");
					row.append("<td>" + result.getString("manoever") + "</td>");
					row.append("<td>" + result.getString("vorsegel") + "</td>");
					row.append("<td>" + result.getString("wdate") + "</td>");
					row.append("<td>" + result.getString("wtime") + "</td>");
					row.append("<td>" + result.getString("motor") + "</td>");
					row.append("<td>" + result.getString("tank") + "</td>");
					row.append("<td>" + result.getString("windstaerke") + "</td>");
					row.append("<td>" + result.getString("windrichtung") + "</td>");
					row.append("<td>" + result.getString("luftdruck") + "</td>");
					row.append("<td>" + result.getString("temperatur") + "</td>");
					row.append("<td>" + result.getString("wolken") + "</td>");
					row.append("<td>" + result.getString("regen") + "</td>");
					row.append("<td>" + result.getString("wellenrichtung") + "</td>");
					row.append("<td>" + result.getString("wellenhoehe") + "</td>");
					row.append("<td style='width:30px; text-align:left;'><div class='btn-group'>");
					row.append("<a class='btn btn-small remove trackingpoint' id='" + result.getString("trackpointnr")
						+ "'><span><i class='icon-remove'></i></span></a>");
					row.append("</div></td>");
					row.append("</tr>");
					
					responseData += row.toString();
				}

			} catch (Exception e) {
			e.printStackTrace();
			}
		}
	return ok(trackingpoint.render(header.render(), header_app.render(), navigation.render("app_trackinginfo"), navigation_app.render("app_trackinginfo"), responseData));
	}
  
	public static Result insert() {
	DynamicForm data = form().bindFromRequest();
	Connection conn = DB.getConnection();
	Statement query;            
	ResultSet result;
	ObjectNode respJSON = Json.newObject();
	int nextId = 0;

	try {
		query = conn.createStatement();

		query.execute("INSERT INTO seapal.trackingPoint (tracknr, lat, lng, windstaerke, windrichtung, luftdruck, temperatur, wolken, regen, wellenhoehe, wellenrichtung, marker, btm, dtm, sog, cog, manoever, vorsegel, wdate, wtime, motor, tank) VALUES ("
		+ data.get("tracknr") + ","
		+ "'" + data.get("lat") + "',"
		+ "'" + data.get("lng") + "',"
		+ "'" + data.get("windstaerke") + "',"
		+ "'" + data.get("windrichtung") + "',"
		+ "'" + data.get("luftdruck") + "',"
		+ "'" + data.get("temperatur") + "',"
		+ "'" + data.get("wolken") + "',"
		+ "'" + data.get("regen") + "',"
		+ "'" + data.get("wellenhoehe") + "',"
		+ "'" + data.get("wellenrichtung") + "',"
		+ "'" + data.get("marker") + "',"
		+ "'" + data.get("btm") + "',"
		+ "'" + data.get("dtm") + "',"
		+ "'" + data.get("sog") + "',"
		+ "'" + data.get("cog") + "',"
		+ "'" + data.get("manoever") + "',"
		+ "'" + data.get("vorsegel") + "',"
		+ "'" + data.get("wdate") + "',"
		+ "'" + data.get("wtime") + "',"
		+ "'" + data.get("motor") + "',"
		+ "'" + data.get("tank") + "');");

		result = query.executeQuery("SHOW TABLE STATUS FROM seapal LIKE 'trackingPoint'");
		if (result.next()) {
			nextId = result.getInt("Auto_increment");
		}
		conn.close();

		respJSON.put("trackpointnr", "" + (nextId - 1));
	} catch (Exception e) {
		respJSON.put("trackpointnr", "Error: " + e);
	}

	return ok(respJSON);
	}

	public static Result delete() {
	DynamicForm data = form().bindFromRequest();
	Connection conn = DB.getConnection();
	Statement query;            
	ResultSet result;
	ObjectNode respJSON = Json.newObject();

	try {
		query = conn.createStatement();
		query.execute("DELETE FROM seapal.trackingpoint WHERE trackpointnr = " + data.get("trackpointnr"));
		conn.close();

		respJSON.put("trackpointnr", "ok");

	} catch (Exception e) {
		respJSON.put("trackpointnr", "Error: " + e);
	}

	return ok(respJSON);
	}
	
}